### PR TITLE
nautilus: core: osd/OSDMap: health alert for non-power-of-two pg_num

### PR DESCRIPTION
**Pull request by smithfarm**
backport tracker: https://tracker.ceph.com/issues/42120

---

backport of https://github.com/ceph/ceph/pull/30525
NOTE: d8d2b71db501da02bb6929aa6e745d832e4cfadb was not included in the backport because `qa/standalone/mon/health-mute.sh` was added post-nautilus
parent tracker: https://tracker.ceph.com/issues/41647

this backport was staged using https://github.com/ceph/ceph/blob/master/src/script/ceph-backport.sh

### CODE REVIEW

**Comment by smithfarm** [comment]
conflicts with #29992 so let's wait for that one to get merged

**Comment by smithfarm** [comment]
Also, in master this caused test failures, so let's wait for a fix to https://tracker.ceph.com/issues/42215 and cherry-pick it in.

**Comment by alfonsomthd** [comment]
@smithfarm @LenzGr @callithea @epuertat This PR should merged together with all related dashboard backend test fixes to avoid "non-power-of-two" warning otherwise 
```jenkins test dashboard backend``` will start failing in nautilus due ti these warnings.

File: PendingReleaseNotes [hunk 1]
```diff
@@ -1,5 +1,19 @@
->=14.2.2
---------
+14.2.5
```
Comment (neha-ojha) [inline PendingReleaseNotes]: We should remember to change this to 14.2.6.

**Comment by neha-ojha** [comment]
@smithfarm this needs a rebase, let's merge this and unblock https://github.com/ceph/ceph/pull/31842

**Comment by smithfarm** [comment]
@neha-ojha How does it look now?

File: src/osd/OSDMap.cc [hunk 2]
```diff
@@ -5857,7 +5856,7 @@ void OSDMap::check_health(CephContext *cct,
       ostringstream ss;
       ss << detail.size() << " pool(s) have non-power-of-two pg_num";
       auto& d = checks->add("POOL_PG_NUM_NOT_POWER_OF_TWO", HEALTH_WARN,
-			    ss.str(), detail.size());
+			    ss.str());
```
Comment (smithfarm) [inline src/osd/OSDMap.cc]: @neha-ojha This appears to be the offending `add()`, but I'm not sure this PR is the right place to fix it?

Comment (smithfarm) [reply src/osd/OSDMap.cc]: ahhh, nevermind - now I see

**Comment by smithfarm** [comment]
@neha-ojha We're expecting another "hotfix" release before the release that will contain this PR, so I wrote "14.2.8" instead of "14.2.7". Otherwise, I made all the changes as you requested. Thanks!

**Comment by neha-ojha** [comment]
> @neha-ojha We're expecting another "hotfix" release before the release that will contain this PR, so I wrote "14.2.8" instead of "14.2.7". Otherwise, I made all the changes as you requested. Thanks!

good to go!

**Comment by yuriw** [comment]
[wip-yuri-testing-2020-01-27-1500-nautilus](https://trello.com/c/ALOoWdsM/816-wip-yuri-testing-2020-01-27-1500-nautilus)

**Pull request merged**